File: my-folder/1146-greatest-common-divisor-of-strings/solution.java
class Solution {
    public String gcdOfStrings(String str1, String str2) {
        	ArrayList<Integer> commonDivisor = new ArrayList<Integer>();
		int a = str1.length();
		int b = str2.length();
		for (int i = Math.min(a, b); i >= 1; i--) {
			if (a % i == 0 && b % i == 0) {
				commonDivisor.add(i);
			}
		}
		String temp;
		for (int i : commonDivisor) {
			temp = str1.substring(0, i);
			boolean divisble = true;
			for (int j = 0; j <= a - i; j += i) {
				if (!str1.substring(j, j + i).equals(temp)) {
					divisble = false;
					break;
				}
			}
			if (!divisble) {
				continue;
			}
			for (int j = 0; j <= b - i; j += i) {
				if (!str2.substring(j, j + i).equals(temp)) {
					divisble = false;
					break;
				}
			}
			if (divisble) {
				return temp;
			}
		}
		return "";
    }
}

File: my-folder/1146-greatest-common-divisor-of-strings/solution.cpp
class Solution {
public:
    bool isDivisor(string str1, string str2)
    {
        int len1 = str1.length();
        int len2 = str2.length();
        string temp;
        for (int i = 0; i < len1 / len2; i++)
        {
            temp.append(str2);
        }
        if (str1.compare(temp) == 0)
        {
            return true;
        }
        return false;
    }
    string gcdOfStrings(string str1, string str2) {
        string result = "";
        int len1 = str1.length();
        int len2 = str2.length();
        if (len1 >= len2)
        {
            for (int i = len2; i > 0; i--)
            {
                if (len1 % i != 0 || len2 % i != 0)
                {
                    continue;
                }
                for (int j = 0; j <= len2 - j; j++)
                {
                    string temp = str2.substr(j, i);
                    //cout << "temp " << temp << endl;
                    if (isDivisor(str1, temp) && isDivisor(str2, temp))
                    {
                        return temp;
                    }
                }
            }
        }
        else
        {
            for (int i = len1; i > 0; i--)
            {
                if (len1 % i != 0 || len2 % i != 0)
                {
                    continue;
                }
                for (int j = 0; j <= len1 - j; j++)
                {
                    string temp = str1.substr(j, i);
                    //cout << "temp " << temp << endl;
                    if (isDivisor(str2, temp) && isDivisor(str1, temp))
                    {
                        return temp;
                    }
                }
            }
        }
        return result;
    }
};
